src/identity-utils.ts

```diff
@@ -0,0 +1,160 @@
+import { Dictionary } from './utils';
+import { LocalStorageVault } from './vault';
+import Types from './types';
+import { IdentityApiData, UserIdentities } from '@mparticle/web-sdk';
+import { isObject } from './utils';
+import { MParticleWebSDK } from './sdkRuntimeModels';
+
+export interface IKnownIdentities extends UserIdentities {
+    device_application_stamp?: string;
+}
+
+export interface ICachedIdentityCall extends UserIdentities {
+    responseText: string;
+    status: number;
+    expireTimestamp: number;
+}
+
+export const cacheIdentityRequest = (
+    method: string,
+    identities: IKnownIdentities,
+    expireTimestamp: number,
+    idCache: LocalStorageVault<Dictionary>,
+    xhr: XMLHttpRequest
+): void => {
```

Comment:
Based on our conversation, I think keeping this as a pure function makes things cleaner. We should remove the dependency on the vault and simply have it return a dictionary that can be inserted into the vault.

For example, the new signature would become:

```typescript
export interface IdentityRequestCacheItem {
    key: string,
    identityRequestCacheItem: ICacheIdentityCall
}

const createIdentityRequestCacheItem = (
    method: string,
    identities: IKnownIdentities,
    expireTimestamp: number,
    httpStatus: number,
    httpResponseText: string
): IdentityRequestCacheItem => {}
```

Reply:
Based on our 1/19 discussion, we will err on not complicating the `identity.js` module and tidying up functions, and keeping thing in TS for easier future refactoring when `identity.js --> identity.ts`